packager/react-packager/src/node-haste/DependencyGraph/HasteMap.js

```diff
@@ -96,12 +96,17 @@ class HasteMap extends EventEmitter {
       return null;
     }
 
-    // If platform is 'ios', we prefer .ios.js to .native.js which we prefer to
+    // If platform is 'ios10', we prefer .ios10.js to .ios.js
+    // which we prefer .ios.js to .native.js which we prefer to
     // a plain .js file.
     let module = undefined;
     if (module == null && platform != null) {
       module = modulesMap[platform];
     }
+    if (module == null && platform != null) {
```

Comment:
it is the same condition, but the above line might have made module not null.  It's not an else if.  It tricked me up, too.   But this is the cascading code.   If the above code modified module, then the second code doesn't run.

I don't know who chose to write this code like this... feels unreadable, but I followed suite :)

"When in Rome"